libs/s25main/figures/noFigure.cpp

```diff
@@ -772,30 +781,27 @@ void noFigure::DrawWalking(DrawPoint drawPt, glArchivItem_Bob* file, unsigned id
     DrawShadow(drawPt, ani_step, GetCurMoveDir());
 }
 
-/// Zeichnet standardmäßig die Figur, wenn sie läuft aus einem bestimmten normalen LST Archiv
 void noFigure::DrawWalking(DrawPoint drawPt, const ResourceId& file, unsigned id)
 {
     const unsigned ani_step = CalcWalkAnimationFrame();
     drawPt = InterpolateWalkDrawPos(drawPt);
 
-    LOADER.GetPlayerImage(file, id + rttr::enum_cast(GetCurMoveDir() + 3u) * 8 + ani_step)
+    LOADER.GetPlayerImage(file, calcWalkFrameIndex(id, GetCurMoveDir(), ani_step))
       ->DrawFull(drawPt, COLOR_WHITE, world->GetPlayer(player).color);
     DrawShadow(drawPt, ani_step, GetCurMoveDir());
 }
 
 void noFigure::DrawWalking(DrawPoint drawPt)
 {
-    // Figurentyp unterscheiden
     switch(job_)
     {
         case Job::PackDonkey:
         {
             const unsigned ani_step = CalcWalkAnimationFrame();
             drawPt = InterpolateWalkDrawPos(drawPt);
 
-            // Esel
-            LOADER.GetMapTexture(2000 + rttr::enum_cast(GetCurMoveDir() + 3u) * 8 + ani_step)->DrawFull(drawPt);
-            // Schatten des Esels
+            LOADER.GetMapTexture(calcWalkFrameIndex(200, GetCurMoveDir(), ani_step))->DrawFull(drawPt);
```

Comment:
Is the start Index here not 2000 instead of 200?

Reply:
Indeed thanks for catching this! I'll test run this anyway a bit later on. Just wanted the CI to pass already to merge it fast